packages/utils/shared/is-object.ts

```diff
@@ -0,0 +1,3 @@
+export function isObject(input: unknown): input is Record<string | number | symbol, unknown> {
+	return typeof input === 'object' && input !== null && !Array.isArray(input);
```

Comment:
As this excludes Array which are technically objects, I wonder if there's a better naming for this function to indicate this exclusion? 

Coincidentally the vue core does also have a very similar `isObject` in their shared package but theirs' do not exclude arrays which personally feels slightly more inline with the naming "is object": https://github.com/vuejs/core/blob/a374d7e6ed973cde7fae36ee82618cf46a8ba68a/packages/shared/src/general.ts#L49-L51

Would something like their `isPlainObject` be more precise here? Seems like their `toTypeString` does catch Array as `[object Array]`, null as `[object Null]` etc.

Reply:
Hmm perhaps `isObject` would still be better compared to our current alternatives I think 😄 Let's keep this naming for now!